xclim/indices/generic.py

```diff
@@ -701,6 +701,7 @@ def temperature_sum(
 
     out = (data - threshold).where(cond).resample(time=freq).sum()
     out = direction * out
+    out.attrs["units_metadata"] = "temperature: difference"
     return to_agg_units(out, data, "integral")
```

Comment:
You mean revert going through absolute temperature ? I *think* this is not necessary with this PR. 

Not sure about `to_agg_units`. Is your logic "if we aggregate temperatures, then it's obviously a difference" ?

Reply:
Well, going to absolute temperatures and this PR both try to solve the same problem. I guess they are compatible, but I feel both aren't needed at the same time. We could revert to make it cleaner.

My point about `to_agg_units` is that that function is there to manage units after a transformation. Instead of going per-indicator, you could simply infer the "units_metadata" needed for the given transformation. Like I did in #1804.

Reply:
I didn't want to sabotage the work you had done and it seemed prudent to keep the changes as small as possible. So no strong opinion on this one. 

My thinking was that `to_agg_units` has no way to know whether it is aggregating a temperature or a temperature difference if we give it a DataArray with units of K.

Reply:
Well it now does/could with your additions, no ? But in any case, isn't the logic the same ? All operations that transform an `on_scale` to a `difference` will also transform a `difference` to a `difference`. I don't think we have operations that would actually go from a `difference` to an `on_scale` ?

Reply:
Ok, removed ensure_absolute_temperature from to_agg_units. Let me know if I misunderstood the intent.